### ISSUE TITLE
Power instabilities

### ISSUE THREAD

**Issue by Staacks** [Staacks/wifi-game-boy-cartridge#2]
The problem has two sides:

- When I turn on the Game Boy, the Nintendo logo is often surrounded by glitches that form a repeating pattern along the tile grid. I think this is due to a voltage drop as the ESP starts while the Game Boys voltage supply is not entirely stabilized, corrupting some of its memory. I think that this would require delaying either the start of the ESP or the Game Boy (via Reset pin?), but this problem is mostly cosmetic as I do not see any problems later and the glitches disappear after the logo screen.
- If the battery voltage is a bit too low (and I am talking about something like 1.48 V per battery), the ESP does not boot reliably. If it is even lower (especially with NiMh batteries), I cannot get the ESP to boot at all. I have not yet measured the actual voltage at the cartridge at this point (yeah, it's a matter of seconds, but I do not have the Game Boy at hand right now - I will do this later), but I think that I underestimated how much the 5V from the Game Boy actually depend on the source voltage and that the dropout voltage of the NCP1117 is just too high to sustain a proper voltage for the ESP in that case. After a quick search I now ordered a few AP7361C-33ER as a pin-compatible replacement with lower drop-out voltage.

**Comment by GeorgeRudolf** [comment]
Just measured on a random DMG here - VBat 3.82V at no load, 3.79V while booting (yes, 4x very worn NiMHs), Vcc at cart slot solid at 5.02V with one of those horrible 25-in-1 carts stabbed in (flashROM+MBC5+SRAM). You might need to add a voltage monitor to the design, so that !RESET is held low until 3.3 is stable. Luckily that's usually just one extra SOT-23.

**Comment by Staacks** [comment]
You are right. I just measured and got 4.95V for all batteries I tried - Also in the case of the NiMh with 4x 1.2V.

However, when I use my WiFi cartridge, the voltage in the latter case drops down to 3V. The Nintendo logo moves properly (although the contrast has to be adjusted in this case) and the ESP just won't start. Is this the load of the ESP (which presumably just keeps "trying" to boot) on the Game Boy in combination with the low voltage from the batteries? I think a delayed start of the Game Boy would only help with the visible memory glitches.

**Comment by Franticware** [comment]
Hi, this is what I am using to stepping down voltage very efficiently:
https://www.aliexpress.com/item/33014445559.html

**Comment by Franticware** [comment]
Hi, this is what I am using to stepping down voltage very efficiently, it should work nicely for this project:
https://www.aliexpress.com/item/32880983608.html
(I suggested this during your youtube Q&A in a form of a question as phrtza)

**Comment by Staacks** [comment]
Thanks. Not exactly a drop-in replacement. :)
I think it is time that I read up on how these things work and how to properly use them instead of just spraying the board with a few capacitors to both sides.

**Comment by Franticware** [comment]
Right, you can't do it without an inductor when you want efficiency. Also please note the row of precision resistors used to select output voltage. Maybe the easiest way is to just buy a few of these modules to harvest the parts. I never looked for schematics nor reverse-engineered the module, but it seems straightforward. I used it as is in my own ESP8266 project to convert power from 7.2V (2x 18650) to 3.3V.
Also, it has enable pin that could theoretically be used for delayed power up of the more power hungry components.
Another note, off topic: You mentioned potential replacement for the EPROM/EEPROM with Flash. Flash chips with parallel interface exist, this one for example is used in Mega Drive flash carts by krikzz: http://www.farnell.com/datasheets/40281.pdf
